staging/src/k8s.io/client-go/discovery/discovery_client_test.go

```diff
@@ -132,15 +132,15 @@ func TestGetServerGroupsWithBrokenServer(t *testing.T) {
 }
 func TestGetServerGroupsWithTimeout(t *testing.T) {
 	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		time.Sleep(2 * time.Second)
+		time.Sleep(8 * time.Second)
```

Comment:
to make it even more resilient to slow test hosts, why not block on a channel here which is closed in an outer defer?

Reply:
Done.